barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -576,10 +576,11 @@ void prove_tube(const std::string& output_path)
     ClientIVC verifier{ builder, input };
 
     verifier.verify(proof);
-    info("num gates in tube circuit: ", builder->get_num_gates());
     using Prover = UltraProver_<UltraFlavor>;
     using Verifier = UltraVerifier_<UltraFlavor>;
     Prover tube_prover{ *builder };
+    // Print the number of gates post finalisation for a precise result
+    info("num gates in tube circuit: ", builder->get_num_gates());
```

Comment:
remove?

Reply:
Erm I guess it's good info generally when running e2e tests

Reply:
note that it calls get_num_gates() and not builder.num_gates. As a sidenote, maybe we can change the name of get_num_gates() to estimate_num_finalized_gates()